### PR TITLE
Sonos now ES6 class plus, two fixes and additional backporting

### PR DESCRIPTION
**Pull request by pascalopitz**
- Sonos now ES6 class rather than prototype syntax
- Added new event 'ContentDirectory'
- Added ContentDirectory service methods
- Fixed mistakes I made with in renderingControl with SetTreble and SetBass, which aren't boolean
- Added getZPInfo method that parses /status/zp and can return householdId, zoneName etc
- Added some service instance shorthands
- Some additional comments

### CODE REVIEW

File: lib/services/RenderingControl.js [hunk 1]
```diff
@@ -111,13 +111,13 @@ class RenderingControl extends Service {
...
   /**
-   * (Un)set bass of a speaker.
-   * @param {boolean} loudness Should it be with or without bass?
+   * Set bass of a speaker.
+   * @param {boolean} bass desired level of bass
    */
   async SetBass (bass) {
     return this._request('SetBass', {
       InstanceID: 0,
-      DesiredBass: (bass ? '1' : '0')
```
Comment (svrooij) [inline lib/services/RenderingControl.js:120]: Are you sure about this? Maybe explain in the comments what the min/max value is.

**Comment by hklages** [comment]
SetBass and SetTreble values should be in range -10 to 10, integer.

**Comment by pascalopitz** [comment]
Yes, quite sure. Also spotted another copy paste mistake I made while trying to confirm to the coding style. I normally don't use comments unless it's an absolute gotcha, so am quite snow blind when it comes to docblocks. Apologies in both instances.

**Comment by svrooij** [comment]
I think this PR is getting pretty big, for us to be able to check it easily I would rather see in broken down in smaller pieces.

- Changing those services (and maybe the extra event)
- Changing **sonos.js**

Because the first part doesn't need a lot of testing, but the second one would require a lot of manual testing (since we don't test every custom method defined there).

An other solution would be for us to release an **alpha** version to npm and let it be tested by various libraries using **node-sonos** and have it tested by use.

What do you guys think @bencevans @hklages

**Comment by pascalopitz** [comment]
Ok, I have broken this up into:

#473 - a fix to my earlier contribution
#474 - Some additional service methods on ContentDirectory, mainly to achieve triggering a share index
#475 - Just a comment
#476 - Event subscription and parsing to ContentDirectory, which enables us to listen to when a share indexing event has started/finished
#477 - This is the rewrite of the main Sonos class to ES6, plus one additional method and two new service shorthands

**Pull request closed without merging**